store/store.go

```diff
@@ -37,16 +37,19 @@ var ErrNotFound = errors.New("eds not found in store")
 
 // Store is a storage for EDS files. It persists EDS files on disk in form of Q1Q4 files or ODS
 // files. It provides methods to put, get and remove EDS files. It has two caches: recent eds cache
-// and availability cache. Recent eds cache is used to cache recent blocks. Availability cache is
-// used to cache blocks that are accessed by sample requests. Store is thread-safe.
+// and availability cache. Recent eds cache is used to cache full EDSes on Put. Availability cache is
+// populated externally by protocols via Cache method. Store is thread-safe.
 type Store struct {
 	// basepath is the root directory of the store
 	basepath string
-	// cache is used to cache recent blocks and blocks that are accessed frequently
-	cache *cache.DoubleCache
+
+	recent       cache.Cache
+	availability cache.Cache
```

Comment:
BUG: They don't share reference counting

* Put A -> Recent
* Get A <- Recent
* Put A -> Avail
* Close A -> Recent
* Get A <- Avail (kaput)

Reply:
Technically this is a bug, but it would never happen in practice because we only put to recent rsmt2d accessor, which has Close as OOO. We should fix this anyway.

Reply:
Once Put starts taking any general Accessor this becomes problematic, so has to be fixed anyway.